pkg/segment/structs/evaluationstructs.go

```diff
@@ -366,7 +366,19 @@ func (self *BoolExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure) (b
 	case "searchmatch":
 		fallthrough
 	case "isnotnull":
-		fallthrough
+		fields := self.GetFields()
+		if len(fields) == 0 {
+			return false, fmt.Errorf("BoolExpr.Evaluate: No fields found for isnotnull operation")
+		}
+
+		value, exists := fieldToValue[fields[0]]
+		if !exists {
+			return false, fmt.Errorf("BoolExpr.Evaluate: Field '%s' not found in data", fields[0])
+		}
+		if value.Dtype != utils.SS_DT_BACKFILL {
+			return true, nil
+		}
+		return false, nil
 	case "isnum":
 		return false, fmt.Errorf("BoolExpr.Evaluate: does support using this operator: %v", self.ValueOp)
 	}
```

Comment:
I think you want to move the `"searchmatch"` case down so the `fallthrough` will hit the `"isnum"` case. Or you can have the `isnum` branch match multiple cases.

Also, you should add a `default` branch here that gives a slightly different error message, like `unknown operator: %v`